common/src/main/java/com/wynntils/features/user/ContainerSearchFeature.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright © Wynntils 2022.
+ * This file is released under AGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.features.user;
+
+import com.wynntils.core.config.Config;
+import com.wynntils.core.features.UserFeature;
+import com.wynntils.gui.render.RenderUtils;
+import com.wynntils.gui.screens.TextboxScreen;
+import com.wynntils.gui.widgets.SearchWidget;
+import com.wynntils.mc.event.ScreenInitEvent;
+import com.wynntils.mc.event.SlotRenderEvent;
+import com.wynntils.mc.objects.CommonColors;
+import com.wynntils.mc.objects.CustomColor;
+import com.wynntils.mc.utils.ComponentUtils;
+import com.wynntils.wynn.item.WynnItemStack;
+import com.wynntils.wynn.item.properties.ItemProperty;
+import java.util.Locale;
+import java.util.regex.Pattern;
+import net.minecraft.client.gui.screens.inventory.AbstractContainerScreen;
+import net.minecraft.world.item.ItemStack;
+import net.minecraftforge.eventbus.api.SubscribeEvent;
+
+public class ContainerSearchFeature extends UserFeature {
+    private static final Pattern BANK_PATTERN = Pattern.compile("§0\\[Pg. (\\d+)\\] §8(.*)'s§0 Bank");
+    private static final Pattern GUILD_BANK_PATTERN = Pattern.compile(".+: Bank \\(.+\\)");
+    private static final Pattern MEMBER_LIST_PATTERN = Pattern.compile(".+: Members");
+
+    @Config
+    public boolean filterInBank = true;
+
+    @Config
+    public boolean filterInGuildBank = true;
+
+    @Config
+    public boolean filterInGuildMemberList = true;
+
+    @Config
+    public CustomColor highlightColor = CommonColors.MAGENTA;
+
+    @SubscribeEvent
+    public void onScreenInit(ScreenInitEvent event) {
+        if (!(event.getScreen() instanceof AbstractContainerScreen<?> screen)) return;
+
+        assert screen instanceof TextboxScreen; // done by mixin
+
+        String title = ComponentUtils.getCoded(screen.getTitle());
+
+        // This is screen.topPos and screen.leftPos, but they are not calculated yet when this is called
+        int renderX = (screen.width - screen.imageWidth) / 2;
+        int renderY = (screen.height - screen.imageHeight) / 2;
+
+        if (filterInBank && BANK_PATTERN.matcher(title).matches()) {
+            addSimpleSearchWidget(screen, renderX, renderY);
+        }
+
+        if (filterInGuildBank && GUILD_BANK_PATTERN.matcher(title).matches()) {
+            addSimpleSearchWidget(screen, renderX, renderY);
+        }
+
+        if (filterInGuildMemberList && MEMBER_LIST_PATTERN.matcher(title).matches()) {
+            addSimpleSearchWidget(screen, renderX, renderY);
+        }
```

Comment:
Extract into a separate method that basically ANDs all three of them

```java
public boolean shouldAddSearch(String title) {
    if (filterInBank && BANK_PATTERN.matcher(title).matches()) {
        return true;
    }

    if (filterInGuildBank && GUILD_BANK_PATTERN.matcher(title).matches()) {
        return true;
    }

    if (filterInGuildMemberList && MEMBER_LIST_PATTERN.matcher(title).matches()) {
        return true;
    }

    return false;

}
```

On that note, pattern matching should be extracted into `/wynn/`

Reply:
Yes and no. We should not just move the patterns; that is pointless. But I have been thinking we'd need like a ContainerType enum in `wynn`, and a `ContainerType identifyContainer(Container c)`. It could also be given a bunch of convenience methods like `isBankPage`, `isLootChest` etc.

The ContainerIdentifier could also possibly add some additional data, like bank page number, loot chest tier, etc.

Reply:
But I agree with the `shouldAddSearch`. As long as the end action is the same, and we don't adapt the  `addSimpleSearchWidget` call, that is clearer.

Reply:
(Also, why `simple`? If there's only one search widget at the moment, you can skip that part of the name)